### ISSUE TITLE
WebDriver tagging to WebDriverListener - Actions class perform() returns with exception

### ISSUE THREAD

**Issue by deeppatel2** [SeleniumHQ/selenium#9770]
## 🐛 Bug Report

When WebDriver instance is tagged/attached to WebDriverListener, using Actions class perform() call results in exception.

Actions class and related methods work fine if WebDriverListner is not used.

## To Reproduce

Detailed steps to reproduce the behavior:
Initialize a WebDriver
Add/attach WebDriverListener
Use Actions class method.

## Expected behavior

Actions class method should work with no exception

## Test script or set of commands reproducing this issue

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.events.EventFiringDecorator;
import org.openqa.selenium.support.events.WebDriverListener;
import org.testng.annotations.Test;

import com.utilities.testng.driver.EventListner;

public class DemoTest2 {

	@Test
	public void demo() throws Exception {
		System.setProperty("webdriver.chrome.driver",
				"chromedriver.exe");
		WebDriver driver = new ChromeDriver();
		WebDriverListener evtListener = new EventListner();
		WebDriver instance = new EventFiringDecorator(evtListener).decorate(driver);

		instance.get("https://www.browserstack.com/");
		Actions action = new Actions(instance);
		WebElement element = instance.findElement(By.linkText("Get started free"));
		action.click(element).perform();
		Thread.sleep(10000);
		driver.close();
	}
}

## Exception Trace

org.openqa.selenium.json.JsonException: java.lang.reflect.InvocationTargetException
Build info: version: '4.0.0-beta-4', revision: '29f46d02dd'
System info: host: 'PF236H45', ip: '192.168.68.63', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '16.0.1'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [8da30166233e9c8b4ccc13ad019e6eb5, actions {actions=[org.openqa.selenium.interactions.Sequence@2b06681c]}]
Capabilities {acceptInsecureCerts: true, browserName: chrome, browserVersion: 92.0.4515.159, chrome: {chromedriverVersion: 92.0.4515.107 (87a818b10553..., userDataDir: C:\Users\depatel\AppData\Lo...}, goog:chromeOptions: {debuggerAddress: localhost:50969}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:50969/devtoo..., se:cdpVersion: 92.0.4515.159, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: accept and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: 8da30166233e9c8b4ccc13ad019e6eb5
	at org.openqa.selenium.json.SimplePropertyDescriptor.lambda$getPropertyDescriptors$1(SimplePropertyDescriptor.java:107)
	at org.openqa.selenium.json.JsonOutput.mapObject(JsonOutput.java:392)
	at org.openqa.selenium.json.JsonOutput.lambda$new$31(JsonOutput.java:207)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.mapObject(JsonOutput.java:395)
	at org.openqa.selenium.json.JsonOutput.lambda$new$31(JsonOutput.java:207)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.lambda$new$24(JsonOutput.java:182)
	at java.base/java.util.HashMap.forEach(HashMap.java:1425)
	at org.openqa.selenium.json.JsonOutput.lambda$new$25(JsonOutput.java:177)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.lambda$new$22(JsonOutput.java:169)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1242)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.openqa.selenium.json.JsonOutput.lambda$new$23(JsonOutput.java:169)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.lambda$new$24(JsonOutput.java:182)
	at java.base/java.util.HashMap.forEach(HashMap.java:1425)
	at org.openqa.selenium.json.JsonOutput.lambda$new$25(JsonOutput.java:177)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.convertUsingMethod(JsonOutput.java:360)
	at org.openqa.selenium.json.JsonOutput.lambda$new$16(JsonOutput.java:154)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.lambda$new$22(JsonOutput.java:169)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.openqa.selenium.json.JsonOutput.lambda$new$23(JsonOutput.java:169)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.lambda$new$24(JsonOutput.java:182)
	at com.google.common.collect.SingletonImmutableBiMap.forEach(SingletonImmutableBiMap.java:66)
	at org.openqa.selenium.json.JsonOutput.lambda$new$25(JsonOutput.java:177)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:287)
	at org.openqa.selenium.json.JsonOutput.write(JsonOutput.java:278)
	at org.openqa.selenium.json.Json.toJson(Json.java:42)
	at org.openqa.selenium.remote.codec.AbstractHttpCommandCodec.encode(AbstractHttpCommandCodec.java:253)
	at org.openqa.selenium.remote.codec.AbstractHttpCommandCodec.encode(AbstractHttpCommandCodec.java:127)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:172)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:162)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:137)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:612)
	at org.openqa.selenium.remote.RemoteWebDriver.perform(RemoteWebDriver.java:680)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.openqa.selenium.support.decorators.WebDriverDecorator.call(WebDriverDecorator.java:228)
	at org.openqa.selenium.support.decorators.DefaultDecorated.call(DefaultDecorated.java:48)
	at org.openqa.selenium.support.decorators.WebDriverDecorator.lambda$createProxy$0(WebDriverDecorator.java:287)
	at jdk.proxy2/jdk.proxy2.$Proxy34.perform(Unknown Source)
	at org.openqa.selenium.interactions.Actions$BuiltAction.perform(Actions.java:617)
	at org.openqa.selenium.interactions.Actions.perform(Actions.java:573)
	at DemoTest3.demo(DemoTest3.java:22)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:794)
	at org.testng.TestRunner.run(TestRunner.java:596)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
	at org.testng.SuiteRunner.run(SuiteRunner.java:276)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
	Suppressed: org.openqa.selenium.json.JsonException: Attempting to close incomplete json stream
Build info: version: '4.0.0-beta-4', revision: '29f46d02dd'
System info: host: 'PF236H45', ip: '192.168.68.63', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '16.0.1'
Driver info: driver.version: RemoteWebDriver
		at org.openqa.selenium.json.JsonOutput.close(JsonOutput.java:303)
		at org.openqa.selenium.json.Json.toJson(Json.java:40)
		... 46 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.openqa.selenium.json.SimplePropertyDescriptor.lambda$getPropertyDescriptors$1(SimplePropertyDescriptor.java:105)
	... 94 more
Caused by: org.openqa.selenium.WebDriverException: unknown error: network connection must be enabled
  (Session info: chrome=92.0.4515.159)
Build info: version: '4.0.0-beta-4', revision: '29f46d02dd'
System info: host: 'PF236H45', ip: '192.168.68.63', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '16.0.1'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [8da30166233e9c8b4ccc13ad019e6eb5, getNetworkConnection {}]
Capabilities {acceptInsecureCerts: true, browserName: chrome, browserVersion: 92.0.4515.159, chrome: {chromedriverVersion: 92.0.4515.107 (87a818b10553..., userDataDir: C:\Users\depatel\AppData\Lo...}, goog:chromeOptions: {debuggerAddress: localhost:50969}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:50969/devtoo..., se:cdpVersion: 92.0.4515.159, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: accept and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: 8da30166233e9c8b4ccc13ad019e6eb5
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:200)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:133)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:53)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:184)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:162)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:137)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:612)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:667)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:671)
	at org.openqa.selenium.remote.RemoteExecuteMethod.execute(RemoteExecuteMethod.java:36)
	at org.openqa.selenium.remote.mobile.RemoteNetworkConnection.getNetworkConnection(RemoteNetworkConnection.java:38)
	at org.openqa.selenium.chromium.ChromiumDriver.getNetworkConnection(ChromiumDriver.java:184)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.openqa.selenium.support.decorators.WebDriverDecorator.call(WebDriverDecorator.java:228)
	at org.openqa.selenium.support.decorators.DefaultDecorated.call(DefaultDecorated.java:48)
	at org.openqa.selenium.support.decorators.WebDriverDecorator.lambda$createProxy$0(WebDriverDecorator.java:287)
	at jdk.proxy2/jdk.proxy2.$Proxy34.getNetworkConnection(Unknown Source)
	... 99 more

## Environment

OS: Windows 10
Browser: Chrome
Browser version: 92.0.4515.159
Browser Driver version: 92.0.4515.107

**Comment by pujagani** [comment]
Thank you for providing the details. I was able to reproduce the issue with Selenium 4.0.0 Beta-4. However, it has been fixed as part of 4798b00 and 87fdef6. Thank you @shs96c for the fix!
After trying with the changes from the trunk, I am no longer able to see the issue. It will be available as part of the next release.

I have used a simple logging event listener as follows:
```
public class EventFiringActionTest {

  @Test
  public void eventFiringActionTest() throws InterruptedException {
    WebDriver driver = new ChromeDriver();
    WebDriverListener evtListener = new EventListener();
    WebDriver instance = new EventFiringDecorator(evtListener).decorate(driver);

    instance.get("https://www.browserstack.com/");
    Actions action = new Actions(instance);
    WebElement element = instance.findElement(By.linkText("Get started free"));
    action.click(element).perform();
    Thread.sleep(10000);
    driver.close();
  }

  public static class EventListener implements WebDriverListener {

    final Logger logger = LoggerFactory.getLogger(Thread.currentThread().getName());

    @Override
    public void beforeGet(WebDriver driver, String url) {
      logger.info("before {}", url);
    }

    @Override
    public void afterGet(WebDriver driver, String url) {
      logger.info("after {}", url);
    }

    @Override
    public void beforePerform(WebDriver driver, Collection<Sequence> actions) {
      logger.info("Before performing actions");
      actions.stream().forEach(action -> {
        logger.info(action.toJson().toString());
      });
    }

    @Override
    public void afterPerform(WebDriver driver, Collection<Sequence> actions) {
      logger.info("After performing actions");

      actions.stream().forEach(action -> {
        logger.info(action.toJson().toString());
      });
    }
  }
}
```


The result is as follows:
```
Starting ChromeDriver 91.0.4472.101 (af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}) on port 6176
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
[1630300102.551][WARNING]: This version of ChromeDriver has not been tested with Chrome version 92.
Aug 30, 2021 10:38:22 AM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
Aug 30, 2021 10:38:22 AM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 92
Aug 30, 2021 10:38:22 AM com.company.EventFiringActionTest$EventListener beforeGet
INFO: before https://www.browserstack.com/
Aug 30, 2021 10:38:23 AM com.company.EventFiringActionTest$EventListener afterGet
INFO: after https://www.browserstack.com/
Aug 30, 2021 10:38:24 AM com.company.EventFiringActionTest$EventListener beforePerform
INFO: Before performing actions
Aug 30, 2021 10:38:24 AM com.company.EventFiringActionTest$EventListener lambda$beforePerform$0
INFO: {id=default mouse, type=pointer, parameters={pointerType=mouse}, actions=[{duration=100, x=0, y=0, type=pointerMove, origin=[[ChromeDriver: chrome on MAC (77ece3192fe7e9ef426f4c821d9b1dc9)] -> link text: Get started free]}, {button=0, type=pointerDown}, {button=0, type=pointerUp}]}
Aug 30, 2021 10:38:26 AM com.company.EventFiringActionTest$EventListener afterPerform
INFO: After performing actions
Aug 30, 2021 10:38:26 AM com.company.EventFiringActionTest$EventListener lambda$afterPerform$1
INFO: {id=default mouse, type=pointer, parameters={pointerType=mouse}, actions=[{duration=100, x=0, y=0, type=pointerMove, origin=[[ChromeDriver: chrome on MAC (77ece3192fe7e9ef426f4c821d9b1dc9)] -> link text: Get started free]}, {button=0, type=pointerDown}, {button=0, type=pointerUp}]}

===============================================
Default Suite
Total tests run: 1, Passes: 1, Failures: 0, Skips: 0
===============================================
```

**Comment by pujagani** [comment]
I would recommend please try and use the pre-release jar available as part of https://github.com/SeleniumHQ/docker-selenium/releases/tag/4.0.0-rc-1-prerelease-20210823. It has the latest changes. If you still face an issue, we can re-open this issue and triage further as required. Thank you!

**Comment by deeppatel2** [comment]
Thanks, this is fixed as part of 4.0.0-rc-1.
